library/include/borealis/sidebar.hpp

```diff
@@ -68,19 +69,22 @@ class SidebarItem : public View
     void setAssociatedView(View* view);
     View* getAssociatedView();
 
+    void   setAssociatedViewUserData(void* userdata);
+    void*  getAssociatedViewUserData();
+
     ~SidebarItem();
 };
 
 // TODO: Add a style with icons, make it collapsible?
-class Sidebar : public BoxLayout
+class Sidebar : public List
```

Comment:
Should Sidebar even exist? Isn't it only a List with different styling and the active thing? Could it not be all merged into List?

Reply:
Asked the same question myself, how should I tackle this?

1) Remove class entirely and apply margins via a new method inside List
2) Remove class entirely and apply margins via an overloaded constructor inside List
3) Remove class entirely and apply margins via TabFrame
4) Keep Sidebar to override constructor from List to apply margins

Reply:
I'd say a mix of 1 and 4

- Empty Sidebar class entirely and apply margins via a new method inside List
- Call the new margins method in the Sidebar constructor

Reply:
I think I got it, kinda tricky had to backport several things, but it works amazing